libraries/fc/include/fc/io/datastream.hpp

```diff
@@ -118,7 +119,7 @@ class datastream<Streambuf, typename std::enable_if_t<std::is_base_of_v<std::str
       buf.pubseekoff(off, std::ios::beg);
       return true;
    }
-   bool remaining() { return buf.in_avail(); }
+   bool remaining() { return buf.in_avail() >= 0; }
```

Comment:
This may still be wrong if `in_avail` returns 0.  Use `buf.sgetc() != EOF`.

Reply:
Taking your word for it, I didn't investigate this one